source/Module/STIG/Functions.BackupRevert.ps1

```diff
@@ -0,0 +1,581 @@
+    @{N='DesiredValue';E={$_.DesiredValue}},`
+    @{N='Operator';E={$_.Operator}}, `
+    @{N='MitigationTarget';E={$_.MitigationTarget}}, `
+    @{N='MitigationType';E={$_.MitigationType}}, `
+    @{N='MitigationName';E={$_.MitigationName}}, `
+    @{N='MitigationValue';E={$_.MitigationValue}}, `
+    @{N='Key';E={$_.Key}}, `
+    @{N='ValueType';E={$_.ValueType}}, `
+    @{N='ValueName';E={$_.ValueName}}, `
+    @{N='ValueData';E={$_.ValueData}}, `
+    @{N='TargetType';E={$_.TargetType}}, `
+    @{N='Ensure';E={$_.Ensure}}, `
+    @{N='OptionValue';E={$_.OptionValue}}, `
+    @{N='StartupType';E={$_.StartupType}}, `
+    @{N='State';E={$_.State}}, `
+    @{N='Policy';E={$_.Policy}}, `
+    @{N='Identity';E={$_.Identity}}, `
+    @{N='AclObject';E={$_.AclObject}}, `
+    @{N='NoRecursion';E={$_.NoRecursion}}, `
+    @{N='EventLogLevel';E={$_.EventLogLevel}}, `
+    @{N='Path';E={$_.Path}}, `
+    @{N='IdentityReference';E={$_.IdentityReference}}, `
+    @{N='IsInherited';E={$_.IsInherited}}, `
+    @{N='AccessControlType';E={$_.AccessControlType}}, `
+    @{N='FileSystemRights';E={$_.FileSystemRights}}, `
+    @{N='RegistryRights';E={$_.RegistryRights}}, `
+    @{N='InheritanceFlags';E={$_.InheritanceFlags}}, `
+    @{N='PropagationFlags';E={$_.PropagationFlags}}, `
+    @{N='Store';E={$_.Store}}, `
+    @{N='Thumbprint';E={$_.Thumbprint}} `
+    | Export-Csv -NoTypeInformation -Path $path
+}
+
+<#
+    .SYNOPSIS
+        This function is used to revert to the system state at backup time.
+
+    .DESCRIPTION
+        This function utilizes the set method of Invoke-Dscresource to set the system settings using the backup.
+
+    .PARAMETER BackupLocation
+        Specifies the location to store the backup file.
+
+    .PARAMETER StigName
+        Specifies the STIG name used for backup. The name of the STIG can be found under PowerSTIG/StigData/Processed.
+
+    .NOTES
+        This function is meant for use in a development environment
+
+    .EXAMPLE
+        Restore-StigSettings -StigName "WindowsServer-2019-MS-2.2.xml"
+#>
+function Restore-StigSettings
+{
+    [CmdletBinding(SupportsShouldProcess,ConfirmImpact = 'high')]
+    param
+    (
+        [Parameter()]
+        [ValidateScript({Test-Path -Path $_})]
+        [string]
+        $BackupLocation = $ENV:TEMP,
+
+        [Parameter(Mandatory = $true)]
+        [string]
+        $StigName
+    )
+
+    $xmlPath = Join-Path -Path $PSScriptRoot -ChildPath '..\..\StigData\Processed'
+    $exclusion = @("*.org.default.xml","RHEL*","Ubuntu*","Vsphere*")
+    $validStigs = Get-ChildItem $xmlPath -Exclude $exclusion
+
+    if ($validStigs.Name -notContains $StigName.Trim())
+    {
+        $errorArray = $validStigs.Name -join("`n")
+        Write-Host "StigName not valid, options are :`n$errorArray"
+        break
+    }
+
+    # Remove DSC Document to revert system state
+    if($PSCmdlet.ShouldProcess('localhost','Remove-DscConfigurationDocument'))
```

Comment:
I tested the code and all seems well.  So two things that should change here is the ShouldProcess conditional and scope.  What currently happens is the user is prompted for only removing the DSC config, when it should really include the entire function operation, since what's happening below this statement are systemwide changes.  The conditional should have Restore-StigSettings instead of Remove-DscConfig..., the scope should include the foreach loop as well, i.e.:
![Screenshot 2021-07-30 123342](https://user-images.githubusercontent.com/12042836/127684025-fd16e6a2-ffb4-449a-ac8f-b79de0d4560f.jpg)
Also, a space after your keyword "if".